lib/driver.js

```diff
@@ -484,6 +482,45 @@ export class XCUITestDriver extends BaseDriver {
     }
   }
 
+  /**
+   * Handles MJPEG server-related capabilities
+   * @returns {Promise<void>}
+   */
+  async handleMjpegOptions() {
+    await this.allocateMjpegServerPort();
+    // turn on mjpeg stream reading if requested
+    if (this.opts.mjpegScreenshotUrl) {
+      this.log.info(`Starting MJPEG stream reading URL: '${this.opts.mjpegScreenshotUrl}'`);
+      this.mjpegStream = new mjpeg.MJpegStream(this.opts.mjpegScreenshotUrl);
+      await this.mjpegStream.start();
+    }
+  }
+
+  async allocateMjpegServerPort() {
+    const mjpegServerPort = this.opts.mjpegServerPort || DEFAULT_MJPEG_SERVER_PORT;
+    this.log.debug(
+      `Forwarding MJPEG server port ${mjpegServerPort} to local port ${mjpegServerPort}`
+    );
+    try {
+      await DEVICE_CONNECTIONS_FACTORY.requestConnection(this.opts.udid, mjpegServerPort, {
+        devicePort: mjpegServerPort,
+        usePortForwarding: this.isRealDevice(),
+      });
+      } catch (error) {
+        if (_.isUndefined(this.opts.mjpegServerPort)) {
+          this.log.warn(`Cannot ensure MJPEG broadcast functionality by forwarding the ${DEFAULT_MJPEG_SERVER_PORT} ` +
```

Comment:
Please change the message to 

`Cannot forward the device port XXX to the local port XXX. Certain features, like MJPEG-based screen recording, will be unavailable during this session. Try to customize the value of 'mjpegServerPort' capability as a possible solution`